File: src/euler/js/problem_2.js
var _ = require("../../../third_party/underscore/underscore.js")._;

var CEIL = 4000000;

exports.preamble = ["Finding the sum of all the even Fibonacci below",
  CEIL + "."];

exports.solutions = {
  brute_force: function () {
    var fLast = 1;
    var fTotal = 2;
    var sum = 0;
    while (fTotal < CEIL) {
      if (fTotal % 2 == 0) {
        sum += fTotal;
      }
      var next = fTotal;
      fTotal = fLast + fTotal;
      fLast = next;
    }
    return sum;
  }
};

File: src/euler/go_euler/problem_2.go
package go_euler

// Sum even Fibonacci numbers.
// 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...
// -> 2 + 8 + 34 + ...
func Problem2(ceil int) int {
	sum := 0
	c := Fibonacci()

	next := <-c
	for next < ceil {
		if next%2 == 0 {
			sum += next
		}
		next = <-c
	}

	return sum
}
